src/test/util/event_buffer.rs

```diff
@@ -305,12 +275,25 @@ impl<'a, T: Clone> EventSubscriber<'a, T> {
         None
     }
 
-    /// Consume and pass events to the provided closure until it returns Some or the timeout is hit.
-    pub(crate) async fn filter_map_event<F, R>(
-        &mut self,
-        timeout: Duration,
-        mut filter_map: F,
-    ) -> Option<R>
+    /// Get the next unread event from the underlying buffer, waiting for a new one to arrive if all
```

Comment:
I moved this after `try_next` so reading down the file goes in the order of increasing layers.